### DISCUSSION TITLE
Add a delay input

### DISCUSSION THREAD

**Discussion by ColeWelsome** [jonaro00/AutoTrader#4] [Ideas]
Hey @jonaro00,

Just wanted to say thanks! Both your instructions and using the script was very clear.

In case you care, I was running into an issue where your script occasionally broke. I'm not sure what caused it but I suspect the issue lies with my older Galaxy S9 hanging on one of the steps for to long. I haven't tried to figure out exactly which step hung because I manged to get around this by adjusting the `SLEEP_DELAYS` constant. I just added 5 seconds to each value and it seems to have resolved itself.

Perhaps you could add a user input in the command line that would let the user tack on some delay to each of the values in the `SLEEP_DELAYS` constant? I'd write the bit of code myself but I'm unfamiliar with Python.

Just a thought, it's likely not needed in the majority of cases.

Thanks again,
-C

**Comment by jonaro00** [comment]
Hi Cole,

Thanks for the good suggestion. I have now added a command `delay` for showing and modifying a custom delay modifier to the sleep durations.

```
Number of trades? ('q' to quit) > delay
Current extra delay: 0

Number of trades? ('q' to quit) > delay 3.5
Current extra delay: 3.5
```

I've made it only apply to the buttons that need to wait for the game server. The First Pokemon button and X button remain unaffected (1 second) since they should always complete in the same amount of time. If you still want those delays to apply there, you can go and flip `False` to `True` in the `BUTTONS` definition.

Feel free to test it out and let me know if it worked or needs any other improvements.

**Comment by ColeWelsome** [comment]
Awesome work. Thank you for the changes!

It seemed to work well. I did only test out on 6 trades because I don't have a ton of stuff to trade at the moment.  I need another community day. I'll let you know here how it goes the next time I do a big batch trade.
